sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,268 @@
+
+        if (isSupportedNoneParameterizedType(type)) {
+            return buildNoneParameterizedType(classField, logger);
+        }
+        if (isArrayOrList(type)) {
+            return buildCollectionField(classField, classChain, logger);
+        }
+        if (CLASS_FIELD_HASH_MAP.containsKey(type)) {
+            return CLASS_FIELD_HASH_MAP.get(type);
+        }
+        List<Field> childFields = build((Class<?>)type, classChain, logger);
+        Field searchField = convertToBasicSearchField(classField, logger);
+        searchField.setFields(childFields);
+        searchField.setSearchable(false); // TODO
+        searchField.setFilterable(false); // TODO
+        CLASS_FIELD_HASH_MAP.put((Class<?>) type, searchField);
+        return searchField;
+    }
+
+    private static boolean isSupportedNoneParameterizedType(Type type) {
+        return SUPPORTED_NONE_PARAMETERIZED_TYPE.containsKey(type);
+    }
+
+    private static Field buildNoneParameterizedType(java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        Field searchField = convertToBasicSearchField(classField, logger);
+        return enrichWithAnnotation(searchField, classField, logger);
+    }
+
+
+    private static boolean isArrayOrList(Type type) {
+        if (type.getClass().isArray()) {
+            return true;
+        }
+        return isList(type);
+    }
+
+    private static boolean isList(Type type) {
+        if (type instanceof ParameterizedType) {
+            Type rawType = ((ParameterizedType) type).getRawType();
+            return List.class.isAssignableFrom((Class<?>) rawType);
+        }
+        return false;
+    }
+
+    private static Field buildCollectionField(java.lang.reflect.Field classField,
+        Stack<Class<?>> classChain, ClientLogger logger) {
+        Type componentOrElementType = getComponentOrElementType(classField.getGenericType(), logger);
+        if (isSupportedNoneParameterizedType(componentOrElementType)) {
+            Field searchField = convertToBasicSearchField(classField, logger);
+            return enrichWithAnnotation(searchField, classField, logger);
+        }
+        List<Field> childFields = build((Class<?>)componentOrElementType, classChain, logger);
+        Field searchField = convertToBasicSearchField(classField, logger);
+        searchField.setFields(childFields);
+        return searchField;
+    }
+
+    private static Type getComponentOrElementType(Type arrayOrListType, ClientLogger logger) {
+        if (arrayOrListType.getClass().isArray()) {
+            return arrayOrListType.getClass().getComponentType();
+        }
+        if (isList(arrayOrListType)) {
+            ParameterizedType pt = (ParameterizedType) arrayOrListType;
+            return pt.getActualTypeArguments()[0];
+        }
+        throw logger.logExceptionAsError(new RuntimeException("Should not be there"));
+    }
+
+    private static Field convertToBasicSearchField(java.lang.reflect.Field classField,
```

Comment:
`Field` should be `SearchField` per mode renames.

Reply:
It is checked in recursive call.
```
       List<Field> searchFields = classFields.stream()
            .filter(classField -> !classField.isAnnotationPresent(FieldIgnore.class))
            .map(classField -> buildField(classField, classChain, logger))
            .collect(Collectors.toList());

Reply:
We have two Field in builder, one is Java reflection, one is our search field. 
Will have rename changes afterwards to make this clear